examples/basic-infection/README.md

```diff
@@ -183,82 +186,130 @@ infection status:
+In `update_property_counter` the event handler, the counter increments/decrements
+the appropriate status. It also registers a hook that execute after all plans
+for a time t have executed, i.e., `on_period_end`
+
+```rust
+fn init(context){
+    // Calculate the initial state
+    population = parameters::get_parameter(population);
     for i in 0..population {
-        for person_property in context.get_person_property_types().to_list() {
-            for person_property_value in (person_property.to_list()) {
-                person_property_counter.increment(person_property,person_property_value);
-            }
-        }
+        counter::increment(
+            counter.get_person_property_value(i)
+        );
     }
-    context.observe_person_property_event(update_property_counter);
-    context.add_plan(report_periodic_item, time = 0);
+    context::observe_person_property_event(update_property_counter);
+    context::on_period_end(0, report_periodic_item);
 }
 ```
-Methods are implemented for person property counter to increment and decrement the counters. For the periodic report, one method will release the periodic items to a report based on the period defined in parameters. Changes in the person properties are observed and the callback function `update_property_counter` updates the counts for each property.
-```
-fn update_property_counter(context, person_id,
-                           person_property_type,
-                           previous_person_property_value) {
-    person_property_counter.increment(context.get_person_property(person_property_type, person_id));
-    person_property_counter.decrement(person_property_type, previous_person_property_value);
+Methods are implemented for person property counter to increment and decrement
+the counters. Changes in the person properties are
+observed and the callback function `update_property_counter` updates the
+counts for each property:
+
+```rust
+fn handle_infection_status_change(context, person_id, prev, current) {
+    counter::increment(current);
+    counter::decrement(prev);
 }
+```
 
-fn report_periodic_item(context) {
-    // Report all items in the counter
-    for person_property in context.get_person_property_types().to_list() {
-            for person_property_value in (person_property.to_list()) {
-                report_item = {
-                t = context.get_time(),
-                person_property_type = person_property,
-                person_property_value = person_property_value,
-                person_property_counter = person_property_counter.get_counter(person_property, person_property_value),
-                }
-                context.print_report_data(report_item);
-            }
-        }
+When all plans have executed for a given time t, `ixa` calls the on_period_end
+callback to write the report row and schedule the next periodic report:
+
+```rust
+fn report_periodic_item(day, context) {
```

Comment:
Can day be t? I guess we don't really care about 0.5 days, but I'm trying to think whether there will be a case when someone may want a periodic report for less than a day. Perhaps if we have a vector-borne disease model where we want to keep track of number of infectious bites per hour or something like that?

Reply:
yeah for sure, or it could be longer